### PR TITLE
[N/A] Lint imports, tabs, lists

### LINKED ISSUE #56: ESLint New Rules
- [Limit number of props on one line](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-max-props-per-line.md#limit-maximum-of-props-on-a-single-line-in-jsx-reactjsx-max-props-per-line)
- [Standardize stylesheet property order](https://www.npmjs.com/package/stylelint-config-rational-order)
- [Tab vs Space indenting](https://eslint.org/docs/rules/indent)
- New lines for entries in objects and arrays
  - [Object props on separate lines](https://eslint.org/docs/rules/object-property-newline)
  - [Array items on separate lines](https://eslint.org/docs/rules/array-element-newline)
  - [Consistent line breaks in objects](https://eslint.org/docs/rules/object-curly-newline)
  - [Line breaks for opening and closing brackets of array](https://eslint.org/docs/rules/array-bracket-newline)

### PR DESCRIPTION
**Pull request by jacksonrya**
- Alphabetize imports by module name
- Alphabetize imported members of a module
- Use 4 space indents instead of tabs
- Place object and array elements on new lines
- Place brackets and curlies on new line
- Warn/auto-fix newlines when multiple parameters or arguments used with a function

### CODE REVIEW

**Comment by jacksonrya** [comment]
I don't know what the best course of action is for this PR. The CI will fail because none of the files have been linted with the new rules.

I feel linting the whole project for this PR is a bad idea as it will produce a lot of conflicts with tasks in progress.

Maybe we try and establish a guide that says 'When you start work on a file, make a commit of just linting, then begin changes". If a lint commit isn't taken, reviews would be difficult to conduct as the diff would be mixed with feature changes and style changes. If this is the case, we have to turn off the lint CI for this PR.

**Comment by rahulrajaram** [comment]
My thoughts:

> I feel linting the whole project for this PR is a bad idea as it will produce a lot of conflicts with tasks in progress.

Completely agree.

> Maybe we try and establish a guide that says 'When you start work on a file, make a commit of just linting, then begin changes". If a lint commit isn't taken, reviews would be difficult to conduct as the diff would be mixed with feature changes and style changes.

Yes. It's good to not mix styling/indentation changes with functional changes as the diff becomes harder to negotiate with.

> If this is the case, we have to turn off the lint CI for this PR.

Unfortunate, but necessary.

File: .eslintrc.js [hunk 1]
```diff
@@ -23,10 +23,35 @@ module.exports = {
...
+      "typescript": {
+        "alwaysTryTypes": true // always try to resolve types under `<root>@types` directory even it doesn't contain any source code, like `@types/unist`
+      },
+	},
+	"import/external-module-folders": ["node_modules"],
+	"import/internal-regex": "^@*/"
+  },
   rules: {
 		// These rules are in alphabetical order, minus any symbols (i.e. @typescript goes under T).  Please preserve this when adding.
+		'array-bracket-newline': [
```
Comment (rahulrajaram) [inline .eslintrc.js:42]: Did you use tabs here to retain consistency within this block?

Comment (jacksonrya) [reply .eslintrc.js:42]: the entire file's formatting was inconsistent... ironic

i couldn't figure out how to apply the rules to this file, and i didn't want to correct styling on it yet. just wanted to get the right rules down

**Comment by jacksonrya** [comment]
New Linting Rules and Continuous Integration

We're going to have to figure out to adjust our linting script in order to only lint files that have been staged/ changed from master. And also make sure the CI tasks follows a similar script.

That way, after new linting rules have been added, local linting checks won't get angry at files that a dev didn't change. And, the CI won't get angry for the same reasons

**Pull request closed without merging**